src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/environment/EnvironmentQuery.kt

```diff
@@ -0,0 +1,63 @@
+package no.skatteetaten.aurora.gobo.graphql.environment
+
+import com.expediagroup.graphql.spring.operations.Query
+import graphql.schema.DataFetchingEnvironment
+import no.skatteetaten.aurora.gobo.graphql.token
+import no.skatteetaten.aurora.gobo.integration.boober.EnvironmentService
+import no.skatteetaten.aurora.gobo.integration.boober.MultiAffiliationEnvironment
+import no.skatteetaten.aurora.gobo.integration.mokey.ApplicationDeploymentResource
+import no.skatteetaten.aurora.gobo.integration.mokey.ApplicationService
+import no.skatteetaten.aurora.gobo.security.checkValidUserToken
+import org.springframework.stereotype.Component
+
+@Component
+class EnvironmentQuery(
+    val applicationService: ApplicationService,
+    val environmentService: EnvironmentService
+) : Query {
+
+    suspend fun environments(names: List<String>, dfe: DataFetchingEnvironment): List<Environment> {
+        dfe.checkValidUserToken()
+        // TODO "upgrade" token, check group, use "super" token for Boober requests
+
+        val environments = names.flatMap { environmentService.getEnvironments(dfe.token(), it) }
+        val applicationDeployments =
+            applicationService.getApplicationDeployment(environments.flatMap { it.getApplicationDeploymentRefs() })
```

Comment:
Burde denne metoden ha en s på slutten, da den er plural return?

I tillegg, finnes det noe måte å gruppere resultatet her for å unngå å måtte kjøre en "find" lenger nede? Det er vel strengt tatt en O(n^2)? Kunne et kall per env vært et tema? Disse resultatene er vel cacha i mokey uansett?

Reply:
godt poeng, disse resultatene er cachet i mokey. så en request pr environment vil ha veldig lite å si på performance. Og det er kanskje verdt det for enklere parsing av resultatet? Skal teste litt